enhancements/manifestlist/manifestlist-support.md

```diff
@@ -0,0 +1,277 @@
+such as "environment variables", "entry point" or "open ports" among its fields,
+this extra information, therefore, still belongs to the "child images" (`B` or
+`C` for instance).
+
+### User Stories
+
+#### Support for manifest list import
+
+As an OpenShift user
+I want to be able to import manifest lists
+So that I can refer to images supporting multiple platforms
+
+#### Support for manifest list pull through
+
+As an OpenShift user
+I want to be able to pull through manifest lists
+So that I can cache multi platform images within my cluster
+
+#### Support for pushing manifest lists to image registry
+
+As an OpenShift user
+I want to be able to push manifest lists directly to internal registry
+So that I can publish my multi arch images
+
+#### Support for manifest lists during prune
+
+As an OpenShift user
+I want to be able to automatically clean unused manifest lists
+So that I can keep my storage resource usage under control
+
+### Implementation Details/Notes/Constraints
+
+#### On pushing manifests into integrated registry
+
+Integrated registry will need to accept image pushes by sha, not only by tag.
+The difference here is that the registry is not supposed to create an Image
+object every time it sees a push by sha but only when it sees a push by tag.
+On the latter scenario the registry may need to create more than one image
+(if the pushed manifest is of type manifest list).
+
+### Risks and Mitigations
+
+#### Console usage of ImageStreamImport
+
+OpenShift console uses ImageStreamImport to obtain information about images
+in order to present them to users. When dealing with manifest lists not all
+information will be presented back to the console (the Config portion of an
+image is absent in a manifest list). Depending on how manifest lists support
+is going to be implemented in the console a second request to the API may be
+needed to select which, among "child images", the user wants to see detailed
+information.
+
+#### Image registry pull through authorization
+
+The current Image and ImageStream layouts are leveraged by the image registry
+when verifying if one is allowed or not to execute a pull through for a certain
+blob. APIs needed to be adjusted to take into account this extra level of
+indirection.
+
+Before providing the client with content image registry goes through a series
+of checks to verify if a valid reference for the blob (by sha) exists and is
+valid within the namespace (repository). Image registry currently caches this
+information but some changes are going to be needed to validate using the newly
+introduced "child images".
+
+## Design Details
+
+### Proposed Image API changes
+
+Images are indexed (named) in OpenShift by the hash of their content. Currently
+an Image is represented by the following structure:
+
+```go
+type Image struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	DockerImageReference string
+	DockerImageMetadata DockerImage
```

Comment:
`DockerImageMetadata` is `runtime.RawExtension`: https://github.com/openshift/api/blob/a9e731090f5ed361e5ab887d0ccd55c1db7fc633/image/v1/types.go#L41